### PR TITLE
Hotfix/update react version rename unsafe lifecycles

### PR DESCRIPTION
**Pull request by resource11**
Hey @davidtheclark! I did work on bumping some of the dependencies for this package (and did the same for `focus-trap-react` and `react-displace` to remove unsafe lifecycles from this lovely modal component. Do you mind taking a look?

### CODE REVIEW

**Comment by davidtheclark** [comment]
cc @ArfatSalman @rahildar

**Comment by ArfatSalman** [comment]
@davidtheclark @resource11 Can we put check for `titleId` and `titleText` in the `constructor` of `Modal`? This way we can remove the `componentWillMount` call altogether.

**Comment by resource11** [comment]
Moved the check down to the `render()` method.

File: src/react-aria-modal.js [hunk 1]
```diff
@@ -142,7 +134,7 @@ class Modal extends React.Component {
 
     if (props.underlayStyle) {
       for (const key in props.underlayStyle) {
-        if (!props.underlayStyle.hasOwnProperty(key)) continue;
+        if (!props.underlayStyle.includes(key)) continue;
```
Comment (ArfatSalman) [inline src/react-aria-modal.js]: AFAIK `underlayStyle` is an object, and does not have the `includes` prototype method. Could you explain this a little bit more?

File: src/react-aria-modal.js [hunk 2]
```diff
@@ -193,7 +185,7 @@ class Modal extends React.Component {
 
     if (props.dialogStyle) {
       for (const key in props.dialogStyle) {
-        if (!props.dialogStyle.hasOwnProperty(key)) continue;
+        if (!props.dialogStyle.includes(key)) continue;
```
Comment (ArfatSalman) [inline src/react-aria-modal.js]: AFAIK `underlayStyle` is an object, and does not have the `includes` prototype method. Could you explain this a little bit more?

**Comment by ArfatSalman** [comment]
> We'll still need to work on lifecycle methods for two dependencies (react-displace and focus-trap-react), yes, @ArfatSalman?

`focus-trap-react` does not have a `componentWillUnmount` and we can safely put the `componentWillUnmount` of `react-displace` in the constructor. @davidtheclark @resource11

**Comment by wldcordeiro** [comment]
FYI I'm gonna try moving the work of updating the dependencies to another PR so you can isolate the lifecycle work here.

File: src/react-aria-modal.js (hunk 1 above)
Comment (resource11) [reply src/react-aria-modal.js]: Ahhh, this was me trying to get around a test error without fully thinking it through. I'll revert this change, @ArfatSalman

**Comment by resource11** [comment]
I had already started the other dependency work in those libraries, @wldcordeiro, and was submitting PRs for those. If you want to take them on instead, I'm good with that!

File: src/react-aria-modal.js (hunk 2 above)
Comment (resource11) [reply src/react-aria-modal.js]: Same as above... however eslint is unhappy with `hasOwnProperty`.

**Comment by toolness** [comment]
Hello!  It seems the last activity on this PR was over a month ago--is there any way I can help move it forward?

**Comment by AElmoznino** [comment]
Hey y'all! Good work! Any progress on this? Would be awesome to get the warnings gone 😄

File: src/react-aria-modal.js (hunk 2 above)
Comment (rivajunior) [reply src/react-aria-modal.js]: > Same as above... however eslint is unhappy with `hasOwnProperty`.

@resource11

This is a security issue. You can use a function like the following to "make eslint happy".

```javascript
const hasOwnProperty = (object, prop) => Object
  .prototype
  .hasOwnProperty
  .call(object, prop)
```

Comment (resource11) [reply src/react-aria-modal.js]: Thanks, @rivajunior!

**Comment by resource11** [comment]
My apologies for the delay, @ArfatSalman! I got sidetracked with a Covid-inspired layoff and a job search. Made the requested changes and also bumped up some of the dependencies.

**Comment by marlomgirardi** [comment]
> My apologies for the delay, @ArfatSalman! I got sidetracked with a Covid-inspired layoff and a job search. Made the requested changes and also bumped up some of the dependencies.

I'm just not sure about mixing dependency updates and react unsafe method fix in the same PR.

**Comment by marlomgirardi** [comment]
I've done a fix for the componentDidMount at #121. I'm closing this issue but feel free to add separated MR.

**Pull request closed without merging**